src/_pytest/nodes.py

```diff
@@ -79,6 +82,21 @@ def iterparentnodeids(nodeid: str) -> Iterator[str]:
             pos = at + len(sep)
 
 
+def _imply_path(
+    path: Optional[Path], fspath: Optional[LEGACY_PATH]
+) -> Tuple[Path, LEGACY_PATH]:
+    if path is not None:
+        if fspath is not None:
+            assert Path(fspath) == path, f"{fspath} != {path}"
```

Comment:
I feel we should raise some internal error here, because `_imply_path` is used by public API functions like `from_parent`.

Also the message could be improved a bit, how about: 
```python
"Both path and fspath argument provided, but they differ:\n"
f"path = {path}\n"
f"fspath = {fspath}"
```

Reply:
Gentle ping.

Reply:
Seems this assert is no longer required, given there's a `if Path(fspath) != path:` above?

Reply:
resolved in 6877389a790586fbc8f45d97ffc7ce32dd0405e1